lib/Sema/CSSimplify.cpp

```diff
@@ -3730,6 +3730,7 @@ bool ConstraintSystem::repairFailures(
         // 2. "existential" check, which is handled by a missing conformance
         //    fix.
         if ((lhs->is<BoundGenericType>() && rhs->is<BoundGenericType>()) ||
+            (lhs->is<NominalType>() && rhs->is<NominalType>()) ||
```

Comment:
Ah, I see :) 
Should we update the comment above to metion the new condition?

Reply:
I guess, I could also mention nominal there, it's going to be handled by `deep equality` just like generic types...

Reply:
Adjusted!